### ISSUE TITLE
[Input][docs] Missing label impacting WCAG 3.3.2

### ISSUE THREAD

**Issue by chris-gds** [mui/base-ui#50]
### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### Steps to reproduce 🕹

Link to live example:

https://mui.com/base-ui/react-input/

### Current behavior 😯

![input UI element on MUI](https://github.com/mui/material-ui/assets/71266765/fc80be10-fb97-4bbd-8124-a8db359d3fef)

## Missing label

[Placeholder text](https://www.w3.org/WAI/tutorials/forms/instructions/#placeholder-text) provides instructions or an example of the required data format inside form fields that have not yet been edited by the user. Placeholder text is usually displayed with lower colour contrast than text provided by users, and it **disappears** from form fields when users start entering text. If the placeholder text contains instructional information or examples that disappear, it makes it more difficult for users to check their responses before submitting the form.

While placeholder text provides valuable guidance for many users, placeholder text **is not a replacement for labels**. Assistive technologies, such as screen readers, do not treat placeholder text as labels. Moreover, at the time of writing this tutorial, placeholder text is not broadly supported across assistive technologies and not displayed in older web browsers.

Further reading:
https://www.w3.org/WAI/WCAG21/Understanding/labels-or-instructions.html

### Context 🔦

Input element is missing a label

### Your environment 🌎

<details>
  <summary><code>npx @mui/envinfo</code></summary>

```
  Don't forget to mention which browser you used.
  Output from `npx @mui/envinfo` goes here.
```
</details>

**Comment by michaldudak** [comment]
We wanted to show only the input on the demo to keep its code short.
Note that screen readers correctly announce the input as it has the `aria-label` attribute:

<img width="197" alt="image" src="https://github.com/mui/material-ui/assets/4696105/69f7555a-d2be-4d4b-a780-0b79c5848cf0">

**Comment by chris-gds** [comment]
> We wanted to show only the input on the demo to keep its code short.

Sure, sometimes people want to prioritise aesthetics over function. However including an accessible label, improving the input component or the docs could go a long way to help certain users.

>Note that screen readers correctly announce the input as it has the aria-label attribute:

The a11y tree is a really helpful indication but not a replacement for screenreader testing.  Unfortunately I can't provide a comprehensive set of testing videos across the various screen readers and browser combinations.  However you should be able to review some of the confusion a user on NVDA + Firefox faces in the video below:

https://github.com/mui/material-ui/assets/71266765/444e3938-bd7c-499d-ab29-2013de17bbf1

**Comment by michaldudak** [comment]
> The a11y tree is a really helpful indication but not a replacement for screenreader testing

Of course, I agree completely. I did test it in both NVDA on Windows and VoiceOver on Safari.

> However you should be able to review some of the confusion a user on NVDA + Firefox faces in the video below

Could you explain what you mean? The input's label is announced, and it doesn't differ much from the case with a separate label:

<img width="1145" alt="image" src="https://github.com/mui/material-ui/assets/4696105/a3a738c6-3bf1-4853-ba8a-8117478ddcaa">

Here I've created two inputs - one with a separate label element and one labeled with `aria-label`. As you can see in the Speech Viewer (4 last lines), in Firefox they are announced in a similar manner. On Edge (not shown here) they are announced exactly the same.

**Comment by chris-gds** [comment]
>Of course, I agree completely. I did test it in both NVDA on Windows and VoiceOver on Safari.

Sure but that's only two combinations there are more, [for example testing can be found,](https://www.davidmacd.com/blog/is-placeholder-accessible-label.html) it's not fully supported everywhere or it's announced differently depending on the combination (I'm unclear if this requires an update since publish).

![davidmacd testing matrix for screenreaders and input placeholder support showing that it's not fully supported](https://github.com/mui/material-ui/assets/71266765/172cafde-d1fc-4954-9aaa-a5895ac4777c)

Also to returning to the design and UX [using some research](https://www.nngroup.com/articles/form-design-placeholders/)

> it disappears from form fields when users start entering text. 

Disappearing placeholder text strains users’ [short-term memory](http://www.nngroup.com/articles/short-term-memory-and-web-usability/). If the user forgets the hint, which people often do while filling out long forms, he has to delete what he wrote and, in some cases, click away from the field to reveal the placeholder text again. In an ideal world, users would be entirely focused when filling out a form. But in reality, users multitask. They [have different tabs open](https://www.nngroup.com/articles/multi-tab-page-parking/), or they might be pulled away by an email or phone call. For complex tasks, they might have to stop and go retrieve a document or order number. From our [research on mobile usability](http://www.nngroup.com/reports/mobile-website-and-application-usability), we know that mobile users are also frequently distracted and interrupted while using their devices. So, it’s important to help users pick up where they left off.

> If the placeholder text contains instructional information or examples that disappear, it makes it more difficult for users to check their responses before submitting the form.

Without labels, users cannot check their work before submitting a form.
The lack of labels makes it impossible for customers to glance through the form and make sure that their responses are correct. Similarly, browsers that autocomplete form fields may fill in information incorrectly. If there are no labels, or if special instructions are no longer visible, customers must reveal the placeholder text by deleting the text in each field one by one in order verify that it matches the description. Realistically though, many won’t even realize that potential for error, and they won’t make the effort to double check.

When error messages occur, people don’t know how to fix the problem.
If the form has been filled out, but there are no labels or instructions visible outside the form fields, then users have to go back to each field to reveal the description in order to fix the error.

Placeholder text that disappears when the cursor is placed in a form field is irritating for users [navigating with the keyboard](http://www.nngroup.com/articles/keyboard-accessibility/).
People using the Tab key move quickly from field to field, and they don’t stop to study the next field before tabbing to it.

> While placeholder text provides valuable guidance for many users, placeholder text is not a replacement for labels. Assistive technologies, such as screen readers, do not treat placeholder text as labels. Moreover, at the time of writing this tutorial, placeholder text is not broadly supported across assistive technologies and not displayed in older web browsers.

Some more pattern examples and reasoning:
https://www.a11yproject.com/posts/placeholder-input-elements/

**Comment by michaldudak** [comment]
Note that we're not relying on the screen readers' ability to read the placeholder here. We're providing the `aria-label` attribute so the input would be announced correctly even if it had no placeholder.

For sighted users, I agree that an input without a placeholder in a form is an anti-pattern. In our case, we're not building a form, but showing how an input looks on its own.

**Comment by chris-gds** [comment]
> Note that we're not relying on the screen readers' ability to read the placeholder here. We're providing the `aria-label` attribute so the input would be announced correctly even if it had no placeholder.

Some screen readers won't announce the placeholder. So for some users, it's like it's not there. Should the placeholder content be important to the user in some way it's not available to them.

Also:
_[The first rule of ARIA](https://www.w3.org/TR/using-aria/#rule1) use is "If you can use a native HTML element or attribute with the semantics and behavior you require already built in, instead of re-purposing an element and adding an ARIA role, state or property to make it accessible, then do so."_

In this context it's not needed, if the native `label` and recommended pattern was used. 

> For sighted users, I agree that an input without a placeholder in a form is an anti-pattern. 
 
It's more the missing `label` that is an anti-pattern

> In our case, we're not building a form, but showing how an input looks on its own.

An input without a `label`? Maybe the fix is documentation based. I would also ask, if it's being used in the wild without a label.

**Comment by michaldudak** [comment]
> Some screen readers won't announce the placeholder. So for some users, it's like it's not there. 

You're missing my point. The placeholder is irrelevant as we have the `aria-label` attribute for the screen readers. If we had the `<label>` element, some of the screen readers still wouldn't read the placeholder content.

> An input without a label?

Not to search far, the very input we're using to write these comments doesn't have a label and relies on the `aria-label` attribute to be accessible.

**Comment by chris-gds** [comment]
> You're missing my point. The placeholder is irrelevant as we have the aria-label attribute for the screen readers. If we had the <label> element, some of the screen readers still wouldn't read the placeholder content.

I think you're missing mine, `aria` wouldn't be needed in this way if you used a `label`.

_[The first rule of ARIA](https://www.w3.org/TR/using-aria/#rule1) use is "If you can use a native HTML element or attribute with the semantics and behavior you require already built in, **instead of re-purposing an element and adding an ARIA role**, state or property to make it accessible, then do so."_

What's the justification to forgo a `label`? Aesthetics? 

> the very input we're using to write these comments doesn't have a label and relies on the aria-label attribute to be accessible.

Perhaps consider testing with disabled users and gain some feedback to see if your assumptions are correct?

**Comment by michaldudak** [comment]
> What's the justification to forgo a label?

As I mentioned before, this demo is not a representation of a form but shows how an input alone looks like. A label is a separate element and not a part of the input.
This is in contrast to [Material UI's Text Field](https://mui.com/material-ui/react-text-field/) where the label *is* a part of the TextField component. The Base UI's Input is an equivalent of Material UI's Input (or native input type=text or textarea elements).

We use separate label elements wherever we show a form (such as https://mui.com/joy-ui/getting-started/templates/sign-in-side/).

**Comment by chris-gds** [comment]
> As I mentioned before, this demo is not a representation of a form but shows how an input alone looks like. A label is a separate element and not a part of the input.

Are they not they're intrinsically linked? There doesn't seem to be a dedicated `label` component for `base-ui`? 
Additionally as you've mentioned, you're using `aria` to _replace_ the `label`'s function in the demo?

I would still expect the `label` to be present in the example? Much like how an [input is referenced here](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input).

> This is in contrast to [Material UI's Text Field](https://mui.com/material-ui/react-text-field/) where the label is a part of the TextField component.

 Interesting, thanks for this! These have `labels`.

> The Base UI's Input is an equivalent of Material UI's Input (or native input type=text or textarea elements).

But they have `label`s..

> We use separate label elements wherever we show a form (such as https://mui.com/joy-ui/getting-started/templates/sign-in-side/).

Ace, isn’t this `joi-ui`? 

Also if I look at the `joi-ui` code it’s **not** using `aria` to replace the `label`

```<input placeholder="Type in here…" class="MuiInput-input css-1u0jcuo" value="">```

---

Some suggested solutions:

1) It seems like on `base-ui` removing the `aria-label` would be the simple fix as then that would align with what you're saying. That you're treating the `input` as a stand-alone element.  Maybe then updating the docs to talk about `label` use combined with this element and why it's needed.

2) Alternatively you could add a `label` to the `base-ui` docs so that's it's inline with best practice.

**Comment by michaldudak** [comment]
We decided to remove the Input from Base UI. Developers can use the native HTML input directly.
